pkg/lib/size/size.go

```diff
@@ -56,48 +54,31 @@ func httpSize(r io.Reader) (int64, error) {
 	return 0, nil
 }
 
-// Return the original (virtual) size of the file represented by the passed-in reader. This reader
-// is assumed to be a multi-reader, see importer.constructReaders(), and therefore a read loop is
-// necessary to handle partial reads.
-// Note: each read overwrites `buf` therefore the offset we "seek" in the file is variable.
+// Return the original (virtual) size of the file represented by the passed-in reader. This reader is assumed
+// to be a multi-reader, see importer.constructReaders(). ReadFull is used to read from the reader into a small
+// buffer where the qemu header is examined.
+// Note: qemu header format defines the size offset:
+//   24 - 31:   Virtual disk size in bytes
+//   See: https://github.com/zchee/go-qcow2/blob/master/docs/specification.md#header
 func qemuSize(r io.Reader) (int64, error) {
-	// format (defined: https://github.com/zchee/go-qcow2/blob/master/docs/specification.md#header)
-	// ...
-	// 24 - 31:   Virtual disk size in bytes
 	const (
 		sizeOffset = 24 // bytes
 		fieldSize  = 8  // bytes
 		bufSize    = sizeOffset + fieldSize
 	)
 
 	buf := make([]byte, bufSize)
-	var bufLen int
-
-	// the first multi-reader Read returns only 4 bytes (magic number) so a loop is needed
-	for {
-		len, err := r.Read(buf)
-		bufLen += len
-		if err == io.EOF { // unexpected
-			break
-		}
-		if err != nil {
-			return 0, errors.Wrapf(err, "cannot read qemu file")
-		}
-		if bufLen >= bufSize {
-			break
-		}
+	l, err := io.ReadFull(r, buf)
+	if err != nil {
+		return 0, errors.Wrapf(err, "qemu ReadFull error")
 	}
-	if bufLen < bufSize {
-		return 0, errors.Errorf("unexpected qemu buffer length: %d", bufLen)
+	if l < bufSize {
```

Comment:
ReadFull returns an error in this case, so the code in this block is unreachable.